documentation/src/main/asciidoc/user_guide/execute_grid.adoc

```diff
@@ -0,0 +1,189 @@
+==  Executing code in the Grid
+
+The main benefit of a Cache is the ability to very quickly lookup a value
+by its key, even across machines. In fact this use alone is probably the reason
+many users use Infinispan. However Infinispan can provide many more benefits
+that aren't immediately apparent. Since Infinispan is usually used in
+a cluster of machines we also have features available that can help utilize
+the entire cluster for performing the user's desired workload.
+
+NOTE: This section covers only executing code in the grid using an embedded cache,
+if you are using a remote cache you should check out
+<<executing_code_in_the_remote_grid, Executing code in the Remote Grid>>.
+
+=== Cluster Executor
+
+Since you have a group of machines, why not off load execution of code to them.
+The cache manager comes with a nice utility that allows you to
+execute arbitrary code in the cluster. Note this feature requires no Cache to be used.  This
+link:{javadocroot}/org/infinispan/manager/ClusterExecutor.html[Cluster Executor]
+can be retrieved by calling +executor()+ on the `EmbeddedCacheManager`. This executor is retrievable
+in both clustered and non clustered configurations.
+
+NOTE: The ClusterExecutor is specifically designed for executing code where the code is not reliant
+upon the data in a cache and is used instead as a way to help users to execute code easily
+in the cluster.
+
+This manager was built specifically using Java 8 and such has functional APIs in mind, thus all methods take a functional
+inteface as an argument. Also since these arguments will be sent to other nodes they need to be serializable.  We even
+used a nice trick to ensure our lambdas are immediately Serializable.  That is by having the arguments implement both
+Serializable and the real argument type (ie. Runnable or Function).  The JRE will pick the most specific class when
+determining which method to invoke, so in that case your lambdas will always be serializable.
+It is also possible to use an <<_user_defined_externalizers, Externalizer>> to ensure greatest performance.
```

Comment:
Unless you want to add some benchmark results, I wouldn't boast about "greatest performance" :)
In fact, performance with `Externalizer` should be about the same as with the default serialization, it's only `AdvancedExternalizer` that sends an int id instead of the full class name.